### PR TITLE
FEAT: add eggnog mapper types and formats, rebased to follow main.

### PR DESCRIPTION
**Pull request by Keegan-Evans**

### CODE REVIEW

File: q2_types_genomics/per_sample_data/tests/test_format.py [hunk 1]
```diff
@@ -75,6 +77,27 @@ def test_multifasta_manifest_empty(self):
...
+        with self.assertRaisesRegexp(
+                ValidationError,
+                "Missing one or more files for ContigSequencesDirFmt"):
+            fmt_obj.validate()
+
+    def test_contig_seqs_dir_fmt_good(self):
+        dirpath = self.get_data_path('good_contigs/')
+        format = ContigSequencesDirFmt(dirpath, mode='r')
+
+        format.validate()
```
Comment (misialq) [inline q2_types_genomics/per_sample_data/tests/test_format.py:99]: Is it really necessary to restructure the entire contig data directory just to add one test? There is a test already for asserting the "good contigs" (which you commented out below?). The only other test (the one right above) could have simply be done by pointing to some other existing directory with no format-acceptable files?

File: ci/recipe/meta.yaml [hunk 2]
```diff
@@ -22,11 +22,14 @@ requirements:
...
     - q2-types {{ qiime2_epoch }}.*
     - qiime2 {{ qiime2_epoch }}.*
+    - eggnog-mapper >=2.1.10
+    - qiime2 {{ qiime2_epoch }}.*
 
 test:
   requires:
     - coverage
     - pytest-cov
+    - qiime2 {{ qiime2_epoch }}.*
```
Comment (misialq) [inline ci/recipe/meta.yaml:32]: Why is this added here? The tests have worked without it until now.

**Review by misialq — CHANGES_REQUESTED** [review]
Hey @Keegan-Evans, I looked through your changes and one thing that occurred to me is that this PR also has some unrelated updates to the contig formats - in the future, could you please separate such changes into an independent PR? It'd be easier to review as they are not related to functional annotation. Thanks!

Otherwise, see some comments inline.

File: q2_types_genomics/feature_data/__init__.py [hunk 3]
```diff
@@ -8,12 +8,17 @@
...
+
+from ._type import MAG, NOG, OG, KEGG
 from ._transformer import MAGIterator
 
-__all__ = [
-    'MAG', 'MAGSequencesDirFmt', 'MAGIterator'
-]
+__all__ = ['MAG', 'MAGSequencesDirFmt', 'MAGIterator', 'NOG', 'OG', 'KEGG',
+           'OrthologAnnotationDirFmt', 'OrthologFileFmt',
+           ]
```
[the remark this answers is not in the document]
Comment (Keegan-Evans) [reply q2_types_genomics/feature_data/__init__.py:22]: Our linter likes it and it reduces the diff on any future additions 🤷‍♂️, I can change it, but as far as I am concerned it is good.

File: q2_types_genomics/genome_data/__init__.py [hunk 4]
```diff
@@ -10,17 +10,17 @@
...
-__all__ = [
-    'GenomeData', 'Genes', 'Proteins', 'Loci', 'GFF3Format',
-    'GenesDirectoryFormat', 'ProteinsDirectoryFormat', 'LociDirectoryFormat',
-    'IntervalMetadataIterator'
-]
+__all__ = ['GenomeData', 'Genes', 'Proteins', 'Loci', 'GFF3Format',
+           'GenesDirectoryFormat', 'ProteinsDirectoryFormat',
+           'LociDirectoryFormat', 'IntervalMetadataIterator',
+           'OrthologFileFmt', 'Ortholog', 'SeedOrthologDirFmt', 'BLAST6',
+           ]
```
[the remark this answers is not in the document]
Comment (Keegan-Evans) [reply q2_types_genomics/genome_data/__init__.py:24]: Keep changing this way how? I think maybe I missed what your complaint was before.

File: q2_types_genomics/feature_data/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ebolyen) [reply q2_types_genomics/feature_data/__init__.py:22]: The original was better, its indentation is cleaner. Please do as Michal requests :)

File: q2_types_genomics/per_sample_data/_format.py [hunk 5]
```diff
@@ -158,7 +158,5 @@ def bams_path_maker(self, sample_id, genome_id):
     MultiFASTADirectoryFormat,
     MultiMAGSequencesDirFmt,
     ContigSequencesDirFmt,
-    MultiBowtie2IndexDirFmt,
-    BAMDirFmt,
-    MultiBAMDirFmt
+    MultiBowtie2IndexDirFmt, BAMDirFmt, MultiBAMDirFmt
```
[the remark this answers is not in the document]
Comment (ebolyen) [reply q2_types_genomics/per_sample_data/_format.py:161]: There shouldn't be, @Keegan-Evans, use:
```
git checkout main -- q2_types_genomics/per_sample_data/_format.py
```
to reset this file (and ideally similar commands for anything else that didn't need to be touched specifically).

File: q2_types_genomics/per_sample_data/tests/test_format.py [hunk 6]
```diff
@@ -20,6 +19,9 @@
     ContigSequencesDirFmt, MultiBowtie2IndexDirFmt, BAMDirFmt, MultiBAMDirFmt
 )
 
+# from q2_types.sample_data import SampleData
+# from q2_types_genomics.per_sample_data import Contigs
+
```
[the remark this answers is not in the document]
Comment (Keegan-Evans) [reply q2_types_genomics/per_sample_data/tests/test_format.py:24]: Nope 😅

File: q2_types_genomics/per_sample_data/_format.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Keegan-Evans) [reply q2_types_genomics/per_sample_data/_format.py:161]: It was throwing linting errors at one point(for no apparent reason that I understood 😂)

File: q2_types_genomics/genome_data/tests/test_format.py [hunk 7]
```diff
@@ -13,13 +13,49 @@
...
+    def test_seed_ortholog_dir_fmt_collection(self):
+        dirpath = self.get_data_path('ortholog/')
+        fmt = SeedOrthologDirFmt(dirpath, mode='r')
+
+        for relpath, obj in fmt.seed_orthologs.iter_views(OrthologFileFmt):
+            self.assertIsInstance(obj=obj, cls=OrthologFileFmt)
+            obj.validate()
+
+    def test_genes_dirfmt_fa(self):
+        dirpath = self.get_data_path('genes')
```
Comment (misialq) [inline q2_types_genomics/genome_data/tests/test_format.py:57]: A little leftover here? :)

**Review by misialq — APPROVED** [review]
Hey @Keegan-Evans, many thanks for the clean-ups - all looks good! 🥇 

I saw one test which didn't have any actual code in it, so maybe you could still remove that one (see my comment inline) - otherwise feel free to merge, thanks!

**Review by misialq — APPROVED** [review]
Thanks @Keegan-Evans, looks great! 🚀 I'm merging this one!

**Pull request merged**